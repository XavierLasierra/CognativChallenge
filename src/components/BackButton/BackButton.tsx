import React from "react";
import { TouchableOpacity, View } from "react-native";
import { IBackButtonProps } from "../../types/interfaces";

import styles from "./backButton.styles";

export default function BackButton({ actionOnPress }: IBackButtonProps) {
  return (
    <TouchableOpacity
      accessibilityRole={"button"}
      style={styles.backButton}
      onPress={actionOnPress}>
      <View style={[styles.line, styles.lineTop]} />
      <View style={[styles.line, styles.lineBottom]} />
    </TouchableOpacity>
  );
}
