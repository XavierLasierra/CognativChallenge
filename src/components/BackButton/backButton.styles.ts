import { StyleSheet } from "react-native";
import { Colors } from "../../theme";

export default StyleSheet.create({
  backButton: {
    height: 50,
    left: 20,
    position: "absolute",
    top: 30,
    width: 50,
    zIndex: 2,
  },
  line: {
    backgroundColor: Colors.gray,
    borderRadius: 50,
    height: 7,
    position: "absolute",
    width: 25,
  },
  lineBottom: {
    top: 13,
    transform: [{ rotate: "45deg" }],
  },
  lineTop: {
    transform: [{ rotate: "-45deg" }],
  },
});
