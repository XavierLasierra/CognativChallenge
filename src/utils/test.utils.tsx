import React from "react";
import { render } from "@testing-library/react-native";
import StoreContextProvider from "../contexts/StoreContextProvider";

const customRender = (ui: any, initialState?: any, options?: any) => {
  const AllTheProviders = ({ children }: any) => (
    <StoreContextProvider store={initialState}>{children}</StoreContextProvider>
  );

  return render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };
