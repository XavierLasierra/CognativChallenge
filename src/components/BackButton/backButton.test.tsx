import React from "react";
import BackButton from "./BackButton";
import { render, fireEvent, RenderAPI } from "../../utils/test.utils";

describe("Given a BackButton component", () => {
  describe("When it is rendered", () => {
    let actionOnPress: () => void;
    let screen: RenderAPI;
    beforeEach(() => {
      actionOnPress = jest.fn();
      screen = render(<BackButton actionOnPress={actionOnPress} />);
    });

    test("Then should match the snapshot", () => {
      expect(screen).toMatchSnapshot();
    });

    describe("And you press it", () => {
      test("Then actionOnPress should be called", () => {
        const button = screen.getByRole("button");
        fireEvent(button, "press");

        expect(actionOnPress).toHaveBeenCalled();
      });
    });
  });
});
