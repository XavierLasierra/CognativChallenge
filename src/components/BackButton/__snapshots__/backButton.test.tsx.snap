// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BackButton component When it is rendered Then should match the snapshot 1`] = `
<View
  accessibilityLabel="Go back"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 50,
      "left": 20,
      "opacity": 1,
      "position": "absolute",
      "top": 30,
      "width": 50,
      "zIndex": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e0e0e0",
          "borderRadius": 50,
          "height": 7,
          "position": "absolute",
          "width": 25,
        },
        Object {
          "transform": Array [
            Object {
              "rotate": "-45deg",
            },
          ],
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e0e0e0",
          "borderRadius": 50,
          "height": 7,
          "position": "absolute",
          "width": 25,
        },
        Object {
          "top": 13,
          "transform": Array [
            Object {
              "rotate": "45deg",
            },
          ],
        },
      ]
    }
  />
</View>
`;
